api/eth_types.go

```diff
@@ -83,6 +83,9 @@ func (e EthBytes) MarshalJSON() ([]byte, error) {
 	if len(e) == 0 {
 		return json.Marshal("0x")
 	}
+	if len(e) == 1 && e[0] == 0 {
```

Comment:
What case does this deal with? Where do we return a non-nil byte string with a single 0 byte (which is different to an empty byte string).

Reply:
According to [go-ethereum implementation](https://github.com/ethereum/go-ethereum/blob/687e4dc8555b3b53fd8d4a4974b8f8f4b594b552/common/hexutil/json.go#L147-L148), the bigint's zero value should marshal to 0x0. I'll change it accordingly.

Reply:
This should be removed, and the type of R, S, V should be big.Int instead of bytes. The empty value of big.Int should be 0x0 and the empty value of bytes should be 0x